### PR TITLE
Content: Add very weak pirate ship, Scrapper.

### PR DESCRIPTION
**Pull request by 1010todd**
----------------------
**Content (Artwork / Ship)**

scrapper rv5 thumb@2x

## Summary
This PR add a new ship for the pirate which are weaker than Sparrow to allow Sparrow to actually win against something as an interceptor. When people pick an Interceptor as their starting ship it often means they want to fight, not die, and die, and die again to another Heavy Rocket Hawk that one-shots you (also remove it please, or at least spawn them later in the game). Currently the only pirate ship a Sparrow can beat is another Sparrow, and very occasionally Fury and Javelin Wasp if you're lucky or skilled enough. This means Sparrow pilots have next to nothing they can fight and could be very frustrating especially if you've played other space game before where you could do something other than die. Sure mining in one way to survive but it shouldn't be the only practical way. With Scrappers, Sparrow pilots can board and plunder them but not always capture them as Scrappers have 4 bunks and it could be filled making it sometimes difficult to capture for Sparrows.

Scrappers comes in a few variants from barely flying ones equipped with a blaster and not even shield generators to more regular ones with twin blaster or rarer Gatling variant as they'll use whatever they can get their hand on.

Lore wise these ships are pretty much dug up from the junkyard with a bit of repair often used by brand-new pirates just starting out trying to make their living with whatever they can. Being the weakest around unarmed Star Barges and Shuttles would be their primary target.

**TLDR**: Scrapper is a pile of scraps cobbled together for Sparrows to beat.

scrappers

shuttles

<details>
<summary>High Res Images</summary>

Old:

scrapper2@5x

Darker (Current):

scrapper rv5@2x

scrapper rv3 warehouse

</details>

## Testing Done
Flied around fighting pirates, Scrappers spawn quite often and Sparrow can easily beat them one-on-one but can still potentially get overwhelmed if not careful.

## Save File
This save file can be used to play through the new mission content:
Choose Sparrow start and fly around for random pirate spawn.

## Artwork Checklist
 - [x] I updated the copyright attributions, or decline to claim copyright of any assets produced or modified
 - [x] I created a PR to the [endless-sky-assets repo](https://github.com/endless-sky/endless-sky-assets) with the necessary image, blend, and texture assets: https://github.com/endless-sky/endless-sky-assets/pull/112
 - [x] I created a PR to the [endless-sky-high-dpi repo](https://github.com/endless-sky/endless-sky-high-dpi) with the `@2x` versions of these art assets: https://github.com/endless-sky/endless-sky-high-dpi/pull/334

### CODE REVIEW

File: data/human/fleets.txt [hunk 1]
```diff
@@ -1837,6 +1837,39 @@ fleet "Small Southern Pirates"
 		"Clipper (Speedy)"
 	variant 1
 		"Modified Argosy"
+	add variant 10
```
Comment (Hurleveur) [inline data/human/fleets.txt:1840]: you shouldnt use add variant since this isnt an event modification
just use variant 10 etc
same thing for all the others.

**Review by Hurleveur — COMMENTED** [review]
really like the ship and idea just this one small thing

would be neat if we had a pirate start with one of these without debt one day

File: data/human/fleets.txt (hunk 1 above)
Comment (1010todd) [reply data/human/fleets.txt:1840]: Oh. I forgot to remove it when moving from the plugin.

**Review by Hurleveur — APPROVED** [review]
description needs a bit of work I think but thats not my area, just approving the idea and syntax

would be really neat if some of these were found as derelicts too

**Review by beccabunny — APPROVED** [review]
Damn, very nice ship

File: data/human/ships.txt [hunk 2]
```diff
@@ -2938,6 +2938,50 @@ ship "Scout"
...
+		"X2200 Ion Steering"
+		"Hyperdrive"
+		
+	engine -6 30
+	engine 6 30
+	gun 0 -31
+	leak "leak" 60 50
+	explode "tiny explosion" 10
+	explode "small explosion" 5
+	description "Scrapper is less of a ship model but rather a type of old shuttle parts barely held together that some might call a ship."
```
Comment (Zitchas) [inline data/human/ships.txt:2981]:
```suggestion
	description "The Scrapper is less a model of ship, and more a general reference to ships cobbled together from the remains of scrapped shuttles. While Republic authorities prefer to dismiss them as at best an urban legend, and at worst an almost-illegal fabrication of unlicensed shipyards, these ships are where many aspiring captains from poorer planets get their start."
```

File: data/human/variants.txt [hunk 3]
```diff
@@ -2613,6 +2613,92 @@ ship "Scout" "Scout (Speedy)"
 
 
 
+ship "Scrapper" "Scrapper (Minimal)"
+	outfits
+		"Energy Blaster"
+		
+		"nGVF-AA Fuel Cell"
```
Comment (Zitchas) [inline data/human/variants.txt:2620]:
```suggestion
		"KP-6 Photovoltaic Panel"
```

With recent changes to power systems, it is actually feasible to do a battery only build. I'm not sure our AI is smart enough to realize it needs to run away when the battery gets low, though. I think our fighters are smart enough to head back to their carrier, but a standalone ship? Not sure. As such, the next best thing to being a pure-battery setup for a minimalist version would be a single solar panel.

File: data/human/variants.txt [hunk 4]
```diff
@@ -2613,6 +2613,92 @@ ship "Scout" "Scout (Speedy)"
...
+
+
+ship "Scrapper" "Scrapper (Boarding)"
+	outfits
+		"Twin Blaster"
+		
+		"nGVF-AA Fuel Cell"
+		"LP036a Battery Pack"
+		"D14-RN Shield Generator"
+		"Laser Rifle" 4
```
Comment (Zitchas) [inline data/human/variants.txt:2665]:
```suggestion
		"KP-6 Photovoltaic Panel" 6
		"LP036a Battery Pack"
		"Bunk Room"
		"Fragmentation Grenades" 8
		"Laser Rifle"
```

If this version is aimed at boarding, perhaps it should have a little more focus on being able to board effectively? Based on the combo of the plasma engines and the boarding name, I would expect this is either going after tiny ships on its own, or preferably following in the wake of bigger pirates and quietly trying to capture somewhat bigger ships while the bigger pirate is still engaged against other threats.

Note: I left a single laser rifle there on the assumption that this ship is all about agressively boarding... But the captain's going to want a weapon with better defense so that they can retreat when everyone else has been killed, right? Self preservation at its finest.

**Review by Zitchas — CHANGES_REQUESTED** [review]
I love the concept. We've got modified ships (which are mostly tradeoffs but generally a little better) and marauder ships (which are clearly very much improved, typically in the combat department with minor tradeoffs elsewhere). So having something that's the bottom of the heap, modified, but in a "pulled together from bits of scrap into something usable" is a really fun alternate of that.

I've got a few suggestions mainly oriented around doubling down on the Scrapper theme.

One suggestion I wasn't able to put into a suggestion box is to add one or two of these to merchant and smuggler fleets too. Probably quite rare on the regular merchant side, but it would be quite appropriate for smugglers and Serenity-esque "we're not committed to being law-abiding merchants, but we're not actively pirating people either" type captains.

Another suggestion would be to add (or modify) a few of the variants to use solar as well. It is both quite a viable build for short combat giving a bit more space for other things, and fits the scrapper theme very well.

I was a little concerned that it'd look too similar to the shuttle in flight, but having tested it, I think it is sufficiently different.

edit: Price comparison. The Scrapper as presented has a shipyard price of 258k. With the changes proposed here that can be brought down to 231k.

File: data/human/variants.txt (hunk 3 above)
Comment (Hurleveur) [reply data/human/variants.txt:2620]: Even if the AI isnt smart that'd fit the beginner pirate/ship out of nothing and doesnt function properly perfectly

File: data/human/ships.txt [hunk 5]
```diff
@@ -2938,6 +2938,50 @@ ship "Scout"
...
+			"blast radius" 10
+			"shield damage" 80
+			"hull damage" 50
+			"hit force" 140
+	outfits
+		"Twin Blaster"
+		
+		"nGVF-AA Fuel Cell"
+		"LP036a Battery Pack"
+		"D14-RN Shield Generator"
```
[the remark this answers is not in the document]
Comment (Hurleveur) [reply data/human/ships.txt:2969]: I wish we'd have more ships without a shield generator tbh as a small one often is useless so either we get shield capacitators at some point or just no shield regen

File: data/human/ships.txt [hunk 6]
```diff
@@ -2938,6 +2938,50 @@ ship "Scout"
...
+	thumbnail "thumbnail/scrapper"
+	attributes
+		category "Interceptor"
+		"cost" 60000
+		"shields" 450
+		"hull" 520
+		"required crew" 1
+		"bunks" 4
+		"mass" 62
+		"drag" 1.1
```
Comment (Zitchas) [inline data/human/ships.txt:2952]:
```suggestion
		"drag" 1.4
```

This will still go massively faster than a shuttle (Overriding the nav system safeties, I'm guessing?), but not quite so blindly fast. 

As a comparison thing:
The Sparrow, with 176 mass, with 1 chipmunk plasma thruster, has a top speed of 766.
The Scrapper, with 186 mass, with 1 X2700 ion thruster, has a top speed of 752.

Having a heavier, pieced together junk heap of old ships, powered by a low performance ion engine, ship go almost as fast as a brand new fairly modern purpose-built interceptor with decently effective thrusters is... problematic... to me. 

Granted, the shuttle has more engine space than the sparrow, and I don't have a problem with the player being able to push it up to higher speeds than the sparrow. I just don't think it should be stock at such high speeds.

With a drag of 1.4 it is still a fair bit below the shuttle's 1.7, so it'll still be able to "break the safety margins" on the shuttle, but it's not quite as outrageous compared to brand new interceptors. Drag of 1.4 gives it a stock speed of 591, and with greyhound thrusters it can get up to 947 quite easily. I didn't test atomics, but those should have a max speed well over a thousand; so even with a drag of 1.4 it is still going to be a serious contender for racing against other interceptors, fighters, and small ships.

File: data/human/variants.txt (hunk 3 above)
Comment (1010todd) [reply data/human/variants.txt:2620]: At first I was a little worried they'd be too weak to last more than a few seconds, tested and seems like they'd last long enough to disable a Sparrow (assuming the Scrapper didn't die first) with spare to fly for a bit so should be enough for them to jump in, disable a Shuttle, plunder, and leave. I'll apply this suggestion.

**Review by Zitchas — APPROVED** [review]
Alright! This fits the lore hinted at by the presence of modified and marauder ships: Which is to say, that there are people out there capable of modifying and building starships without the backing of major corporations, governments, or shipyards. And I like the fact that it rounds things out with an addition to the bottom end of the spectrum. 

I also like the fact that this would actually be a pretty good stater ship for a pirate campaign. Amusingly, it's a potential trade off for starter mining ship vs the Sparrow: 5 more cargo, but less DPS potential. Take your pick as to which you prefer.

And it's speed is good enough to make hit-and-run tactics a possibility against shuttles and maybe star barges; and definitely good enough that they might show up in races, too. Probably something akin to our present day "ratrods," (aka hotrods built into the body of a car that looks like it was recently living in a scrapyard).

All in all, it's got some good story potential, some interesting starting uses, and it satisfies its primary purpose of adding in some really bargain-basement ships that a Sparrow in the hands of a beginner pilot actually has a decent hope of winning against.

**Review by Saugia — CHANGES_REQUESTED** [review]
Normally I'd be saying "we have too many human ships and adding another needs some serious reasoning behind it" yadda yadda. A pirate shuttle, however, would *technically* be a variant, perhaps a "marauder" esq vessel, so it bypasses some of that. Having it fill a gap in combat definitely solidifies it's place otherwise, and seems very useful for Sparrow players and an early combat stepping stool early on in general. Very happy to see it using the new Twin Blaster too.

Artwork-wise, I just have two things to say. One is that you are using the type of shading you normally use on your own models (forget what it's called), the one where things look a bit more outlined in white glares and in my opinion looks a bit plastic. I would prefer we use the same shading method throughout all our artwork. Now, I know you did the new Hai Railgun the same way, but I felt that looked good for an outfit, and didn't really expect it to be more widely used. 

I know you've said before the varying artists can stand out in the game, but all of them use the same shading besides this. For the sake of consistency, rather than "well other things are different so this is fine", I would prefer you used the same shading method. If it is and I'm just blind (I just woke up), then it's fine. 

I'd have to see it next to the other two shuttles in game too. The thumbnail is amazing. However, currently, even zoomed in the way it is in the preview image for the top view, it looks like a grey gritty blob. I think it could use more contrast.

**Comment by Zitchas** [comment]
I'm not the graphics person, but I'd just like to say that I like the old image better than the new one. The slightly darker and rougher look to it made it more distinguishable from the shuttle, which is important.

**Review by Saugia — APPROVED** [review]
My comments were addressed. It still seems a tiny bit on the gritty side, but it should be fine.

**Comment by Saugia** [comment]
If you want to make it darker grey a bit like Z said, I suppose you could, but you'd have to adjust the shadow contrast to be even darker and I'm not sure that'll look good.

**Comment by 1010todd** [comment]
I think it still looks different enough from the Shuttle even at furthest zoom. Maybe if I have time I might try to adjust it.

**Comment by Zitchas** [comment]
I'm fine with it. I liked the old look better, but I don't object to the new one, so don't slow this down on account of me.

**Comment by McloughlinGuy** [comment]
This would be a good ship for a pirate start, just saying.

File: data/human/ships.txt [hunk 7]
```diff
@@ -2938,6 +2938,49 @@ ship "Scout"
...
+		"X2200 Ion Steering"
+		"Hyperdrive"
+		
+	engine -6 30
+	engine 6 30
+	gun 0 -31
+	leak "leak" 60 50
+	explode "tiny explosion" 10
+	explode "small explosion" 5
+	description "The Scrapper is less a model of ship, and more a general reference to ships cobbled together from the remains of scrapped shuttles. While Republic authorities prefer to dismiss them as at best an urban legend, and at worst an almost-illegal fabrication of unlicensed shipyards, these ships are where many aspiring captains from poorer planets get their start."
```
Comment (Anarchist2) [inline data/human/ships.txt:2980]:
```suggestion
	description "Rather than being a specific model of ship, the Scrappers are instead a family of vessels cobbled together from the remains of destroyed shuttles. On the rare occasion that Republic authorities bother to recognize their existence, Scrappers are treated as almost-illegal, patent-infringing death traps. Despite this, these ships are where many captains (and pirates) from poorer backgrounds get their start."
```

File: data/human/variants.txt [hunk 8]
```diff
@@ -2613,6 +2613,93 @@ ship "Scout" "Scout (Speedy)"
 
 
 
+ship "Scrapper" "Scrapper (Minimal)"
+	outfits
+		"Energy Blaster"
+
```
Comment (Anarchist2) [inline data/human/variants.txt:2619]: Other ship variants don't have line breaks in their definitions.

File: data/human/sales.txt [hunk 9]
```diff
@@ -102,6 +102,7 @@ shipyard "Northern Pirates"
 	"Dagger"
 	"Firebird"
 	"Headhunter"
+	"Scrapper"
```
Comment (Anarchist2) [inline data/human/sales.txt:105]: Alphabetical order.

Comment (1010todd) [reply data/human/sales.txt:105]: I think you meant to put this on Core Pirates shipyard but alright.

File: data/human/ships.txt (hunk 7 above)
Comment (1010todd) [reply data/human/ships.txt:2980]: I'm not too sure about `remains of destroyed shuttles`, feels too violent and specific..? I like the rest of the suggestion though.

**Review by EjoThims — CHANGES_REQUESTED** [review]

File: data/human/ships.txt (hunk 7 above)
Comment (1010todd) [reply data/human/ships.txt:2980]: "decommissioned" sounds very military/official to me. I was thinking about it and maybe "remains of shuttle wrecks", but by definition it's pretty much the same as "destroyed", so maybe just "remains of shuttles" might work? Other words I was thinking of might be "plundered", "stolen", "broken down" which are kinda different but should give the same idea.

Comment (Zitchas) [reply data/human/ships.txt:2980]: Hmmm, how about merely "broken shuttles" ? I agree that decommissioned sounds rather formal, like ships that were intentionally declared out of service and dismantled properly; whereas I envision these as more being made from shuttles that wore out and just happened to end up parked in a junk spaceship lot. Nothing formal, they're just ships that just stopped getting fixed at some point, and then eventually people tore pieces out of them to stick together to make a new ship.

Some probably were destroyed, but that also sounds deliberate, whereas "broken" is just, well, it's the state they are in.

File: data/human/fleets.txt [hunk 10]
```diff
@@ -2004,6 +2028,10 @@ fleet "Small Independent"
 		"Clipper (Heavy)"
 	variant 1
 		"Clipper (Speedy)"
+	variant 6
+		"Scrapper"
+	variant 4
+		"Scrapper (Minimal)"
```
Comment (Hurleveur) [inline data/human/fleets.txt:2034]: these feel too likely to be honest
how about you cut the weight in two and multiply the amount of them by 2?

File: data/human/fleets.txt [hunk 11]
```diff
@@ -1837,6 +1843,24 @@ fleet "Small Southern Pirates"
 		"Clipper (Speedy)"
 	variant 1
 		"Modified Argosy"
+	variant 2
+		"Scrapper"
+	variant 2
+		"Scrapper (Minimal)"
```
Comment (Hurleveur) [inline data/human/fleets.txt:1849]: I don't think spawning 1 of them should be more likely than spawning two, looking at the rest of the variants they have at least 3 sparrows of strength each, this would significantly weaken these fleets (mb thats intended?)

I'd say spawn 2-3 of them, except in a few rare cases? especially if its a weak version of it I can see pirates being stupid enough to attack with it but not alone

they wont rush towards the player anyway because they consider cost (not heroic) and there is no way you cost less than they do, right?

that said, feel free to ignore me this is content, so not my jam

**Review by Hurleveur — COMMENTED** [review]
just would like to make sure we're aware: we're making pirates way weaker than before with these changes, since the ships are so weak, in low quantities, and one of them can replace a clipper or smth like that that's worth about 3 of them

**Review by Saugia — APPROVED** [review]
After discussion in discord, this seems to be a good place for the PR.

File: copyright [hunk 12]
```diff
@@ -908,6 +908,8 @@ License: CC-BY-SA-4.0
 Comment: Detailed by Becca Tommaso (tommasobecca03@gmail.com).
 
 Files:
+ images/ship/scrapper*
+ images/thumbnail/scrapper*
  images/ship/mammoth*
  images/thumbnail/mammoth*
 Copyright: 1010todd
```
Comment (Saugia) [inline copyright:915]:
```suggestion
 images/ship/mammoth*
 images/ship/scrapper*
 images/thumbnail/mammoth*
 images/thumbnail/scrapper*
Copyright: 1010todd
```

**Review by Hurleveur — APPROVED** [review]
